pkg/instancetype/instancetype.go

```diff
@@ -720,12 +721,26 @@ func (m *InstancetypeMethods) findClusterInstancetypeByClient(resourceName strin
 
 func (m *InstancetypeMethods) InferDefaultInstancetype(vm *virtv1.VirtualMachine) (*virtv1.InstancetypeMatcher, error) {
 	if vm.Spec.Instancetype == nil || vm.Spec.Instancetype.InferFromVolume == "" {
-		return nil, nil
+		return vm.Spec.Instancetype, nil
 	}
 	defaultName, defaultKind, err := m.inferDefaultsFromVolumes(vm, vm.Spec.Instancetype.InferFromVolume, apiinstancetype.DefaultInstancetypeLabel, apiinstancetype.DefaultInstancetypeKindLabel)
+
+	// Ignore certain errors if corresponding policy is set
+	var ignoreableInferenceErr *IgnoreableInferenceError
+	if errors.As(err, &ignoreableInferenceErr) {
+		failurePolicy := vm.Spec.Instancetype.InferFromVolumeFailurePolicy
+		if failurePolicy != nil && *failurePolicy == virtv1.IgnoreInferFromVolumeFailure {
+			// Lazy logger construction
+			logger := func() *log.FilteredLogger { return log.Log.Object(vm) }
```

Comment:
This is the expected behavior, without unknown side-effects, therefore I think logging it is redundant.

Reply:
How would you know a failure happended during inference otherwise, let's say when it happens in large quantities? IMO an info log makes sense in this case to not let it go unnoticed completely.

Reply:
I assume that by choosing to ignore failures we wouldn't need to log them. Maybe I would be able to see the whole picture if you can provide with an example even with large quantities, why would I like to ignore failures and log them at the same time?

Reply:
Perhaps to gather an understanding of the usage of the system? Failures shouldn't happen too often, but when they happen, they could be a legitimate mistake by the user too. To debug such cases a log could be helpful. WDYT?

Reply:
In that case I would prefer to fail fast, i.e. reject the VM creation in case the user specified a source from whom we can't infer the data. I find it hard to come up with a scenario where we prefer to learn about these failures rather than act upon them immediately.

Reply:
I would rather think about a mechanism that logs about missing data in the infer sources proactively.

Reply:
In general we prefer the components to be less chatty unless put in debug loglevel. If you still think this log is useful can you please use `log.Log.Object(vm).V(3)` log level? Our default level is `2`. BTW why would you use a lazy constuctor?

Reply:
Yes, by default the behavior is to fail fast. This policy allows to ignore certain errors, which allows the UX of `virtctl` to be a bit nicer by trying to infer instancetypes if possible but not forcing users to inference (e.g. try to but don't fail if not possible). I addressed your comments about the calls to the logger.